FRENCH/src/ch07-01-packages-and-crates.md

```diff
@@ -0,0 +1,153 @@
+*src/lib.rs*, alors le paquet continent une crate de bibliothèque qui a le même
+nom que le paquet, et que *src/lib.rs* est sa crate racine. Cargo transmet les
+fichiers de la crate racine à `rustc` pour compiler la bibliothèque ou le
+binaire.
+
+<!--
+Here, we have a package that only contains *src/main.rs*, meaning it only
+contains a binary crate named `my-project`. If a package contains *src/main.rs*
+and *src/lib.rs*, it has two crates: a library and a binary, both with the same
+name as the package. A package can have multiple binary crates by placing files
+in the *src/bin* directory: each file will be a separate binary crate.
+-->
+
+Dans notre cas, nous avons un paquet qui contient uniquement *src/main.rs*, ce
+qui veut dire qu'il contient uniquement une crate binaire qui s'appelle
+`mon-projet`. Si un paquet contient *src/main.rs* et *src/lib.rs*, il a deux
+crates : une bibliothèque et une binaire, chacune avec le même nom que le
+paquet. Un paquet peut avoir plusieurs crates binaires en ajoutant des fichiers
+dans le répertoire *src/bin* : chaque fichier sera une crate séparée.
+
+<!--
+A crate will group related functionality together in a scope so the
+functionality is easy to share between multiple projects. For example, the
+`rand` crate we used in [Chapter 2][rand]<!-- ignore -- > provides functionality
+that generates random numbers. We can use that functionality in our own
+projects by bringing the `rand` crate into our project’s scope. All the
+functionality provided by the `rand` crate is accessible through the crate’s
+name, `rand`.
+-->
+
+Une crate regroupera plusieurs fonctionnalités associées ensemble dans une
+portée afin que les fonctionnalités soient faciles à partager entre plusieurs
+projets. Par exemple, la crate `rand` que nous avons utilisé dans
+[le chapitre 2][rand]<!-- ignore --> nous fournit une fonctionnalité qui génère
+des nombres aléatoires. Nous pouvons utiliser cette fonctionnalité dans notre
+propre projet en important la crate `rand` dans la portée de notre projet.
+Toutes les fonctionnalités fournies par la crate `rand` seront accessibles via
+le nom de la crate, `rand`.
+
+<!--
+Keeping a crate’s functionality in its own scope clarifies whether particular
+functionality is defined in our crate or the `rand` crate and prevents
+potential conflicts. For example, the `rand` crate provides a trait named
+`Rng`. We can also define a `struct` named `Rng` in our own crate. Because a
+crate’s functionality is namespaced in its own scope, when we add `rand` as a
+dependency, the compiler isn’t confused about what the name `Rng` refers to. In
+our crate, it refers to the `struct Rng` that we defined. We would access the
+`Rng` trait from the `rand` crate as `rand::Rng`.
+-->
+
+Garder une fonctionnalité d'une crate dans sa propre portée clarifie si une
+fonctionnalité précise est définie dans notre crate ou dans la crate `rand` et
+évite ainsi des potentiels conflits. Par exemple, la crate `rand` fournit un
```

Comment:
Could probably be simplified to be easier to understand.

Reply:
Attempting to correct in 6cc61cb6b9cac5688f9243fec4ab5383b00b154c